contribute/design/1.x/cstor/cstor-status.md

```diff
@@ -0,0 +1,220 @@
+```yaml
+Conditions:
+  - lastUpdateTime: 
+    lastTransitionTime: 
+    status: "True"
+    type: Expanded
+
+Conditions:
+  - lastUpdateTime: 
+    lastTransitionTime: 
+    reason: ExpansionInProgress
+    status: "Unknown"
+    type: Expanded
+```
+
+**DiskReplaced** : The DiskReplaced condition gets appended when someone triggers disk replacement on that pool and represents the status of replacement. If multiple disks were replacing then condition message will show that the following are block devices that were under replacement. Further information will be available on corresponding CSPI events.
+```yaml
+Conditions:
+  - lastUpdateTime: 
+    lastTransitionTime: 
+    status: "True"
+    type: DiskReplaced
+
+Conditions:
+  - lastUpdateTime: 
+    lastTransitionTime: 
+    message: ‘error msg from zfs command'
+    reason: ReplacementFailed
+    status: "False"
+    type: DiskReplaced
+```
+
+**DiskUnavailable** : The DiskUnavailable condition gets appended when someone when one or more disk gets into an unavailable state. If multiple disks were unavailable then same DiskUnavailable will set to true. The condition message will have information about the names disks were unavailable. Further information will be available on events of corresponding CSPI.
+```yaml
+Conditions:
+  - lastUpdateTime: 
+    lastTransitionTime: 
+    message: ‘disk gone bad’
+    reason: DiskFailed
+    status: "True"
+    type: DiskUnavailable
+
+Conditions:
+  - lastUpdateTime: 
+    lastTransitionTime: 
+    status: "False"
+    type: DiskUnavailable
+```
+
+**PoolLost** : The PoolLost condition gets appended when the pool import fails because of some reason. 
+```yaml
+Conditions:
+  - lastUpdateTime: 
+    lastTransitionTime: 
+    message: ‘unable to import pool’
+    reason: ImportFailed
+    status: "True"
+    type: PoolLost
+
+Conditions:
+  - lastUpdateTime: 
+    lastTransitionTime: 
+    status: "False"
+    type: PoolLost
+```
+
+#### CSPC Status
+
+The current CSPC does not have any status to represent the state of the CSPC whether all the CSPI got provisioned or not, how many CSPI are healthy or some other state.
+
+The CSPC status should be informative enough to tell the current state of the provisioned instances whether they are in Healthy/Other phase and whether all instances are provisioned or not. It should not be having the details of the instances as the CSPI status already have the corresponding status and repeating the status is not required. The below example shows how the status should look like: 
+```sh
+NAME            PROVISIONEDINSTANCES   RUNNINGINSTANCES   HEALTHYINSTANCES   AGE
+sparse-pool-1           2/3                  1           	      1            142M
+
+NAME            PROVISIONEDINSTANCES   RUNNINGINSTANCES   HEALTHYINSTANCES   AGE
+sparse-pool-1           3/3                  3           	      3            142M
```

Comment:
here also, how CSPC gets these various counts i.e., healthy, running instances?
are there any changes to status of CSPC?